launcher-service/src/controllers/apicontroller.ts

```diff
@@ -267,30 +268,24 @@ export const createJob = async (
     }
 
     if (process.env.ESCROW_CREATION_ENABLED) {
-      const launcherServerEscrowUrl = `${process.env.LAUNCHER_SERVER_URL}/escrow`;
+      const queue = 'CREATE_ESCROW';
+      const rabbitMQconnection = await amqplib.connect('amqp://localhost');
+      const rabbitMQchannel = await rabbitMQconnection.createChannel();
+      await rabbitMQchannel.assertQueue(queue);
 
-      const response = await axios.post(launcherServerEscrowUrl, {
+      const data = {
         chainId: job.group.chainId,
         description,
         fortunesRequired: reviewersRequired,
         fundAmount,
         jobRequester: job.group.creator.address,
         title,
         token: job.group.token,
-      });
-
-      const { escrowAddress } = response.data;
-
-      await prisma.job.update({
-        where: {
-          id: job.id,
-        },
-        data: {
-          escrowAddress,
-        },
-      });
-
-      return res.status(201).json({ job, content, escrowAddress });
+        jobId: job.id,
+      };
+      rabbitMQchannel.sendToQueue(queue, Buffer.from(JSON.stringify(data)));
+      await rabbitMQchannel.close();
+      await rabbitMQconnection.close();
     }
 
     res.status(201).json({ job, content });
```

Comment:
Also, in this endpoint we should take the input in the queue and return the id of the queue.. create new endpoint to provide the status of that queue id..

Do async process for job & content creation include in the queue!!

Reply:
Hey @sarthakvijayvergiya , RabbitMQ doesn't give any queue id after publishing the message. So, I kept the job and content creation as it was.